### ISSUE TITLE
Downloading worse subtitle after succesful 100% embedded extraction

### ISSUE THREAD

**Issue by PCJones** [morpheus65535/bazarr#2170]
**Describe the bug**
Bazarr is downloading subtitles from other providers even though there was a 100% score embedded subtitle that got extracted.

**To Reproduce**
1. Click on 'Wanted'
2. Click on 'Movies' or 'Episodes'
3. Click on 'Search All'

**Expected behavior**
If there is an embedded subtitle bazarr should use that and not download worse subtitles from other providers.

**Screenshots**
![unbenannt](https://github.com/morpheus65535/bazarr/assets/377223/5e0f48ca-f3ce-48af-a1b6-e32bdc0728f0)
![unbenannt](https://github.com/morpheus65535/bazarr/assets/377223/63ff8f8f-f28a-4280-ba9b-2e72827b95f6)

**Software (please complete the following information):**
 - Bazarr: [v 1.2.1]
 - Radarr version [e.g. v 4.0.0.535]
 - Sonarr version [e.g. v 4.5.2.7388]
 - OS: Debian 10 (Linux-5.10.0-23-amd64-x86_64-with)

**Additional context**
I suspect it might be because the subtitle extraction takes quite a long time as it's an SMB drive, however I'm not sure of that. If I deactive all other providers subtitle extraction works like a charm though, it's just slow. It's also interesting to note that - at least in the history - they all show up at the same time (apparently after the embedded extraction is finished)

**Comment by morpheus65535** [comment]
Without a debug showing the sequence of actions, there's not much that can be done here. Bazarr shouldn't download (this isn't an upgrade) subtitles when there's an existing subtitles indexed properly.

**Comment by halali** [comment]
All are downloaded at same time? Isnt that strange?

**Comment by PCJones** [comment]
Single language mode is disabled. Language profile is looking for German, German forced, English, English forced with no cutoff.

I've deleted the episode and all subtitles, cleared the embededsubtitles cache folder and downloaded the episode again.

It looks like this before I click on "Search All", so it seems like the embedded subtitles are not indexed properly yet:
![unbenannt2](https://github.com/morpheus65535/bazarr/assets/377223/b290a35f-cff0-4ad5-9f0a-0ade6cbec172)
The episode does have all required languages embedded so it shouldn't look at other providers at all (German, English, German Forced, English Forced) 

This is what happens:
![Unbenannt](https://github.com/morpheus65535/bazarr/assets/377223/001722f8-7caf-4473-a674-336fde02f63b)
It only seems to extract the german forced embedded subtitle, but it also downloads other german forced subtitles for no apparent reason.

Debug log:
https://gist.github.com/PCJones/a0ce4a78534249b8fa9d4322601a1995
The log file is quite large as I didn't want to delete any information that might be necessary, however the process for the affected episodes start within the first 40 lines.

**Comment by morpheus65535** [comment]
@vitiko98 could it be an issue with embedded subtitles provider and forced subtitles?

```
12/06/2023 12:14:11|INFO    |subliminal_patch.core_persistent|Downloaded 5 subtitle(s)|
12/06/2023 12:14:11|DEBUG   |subliminal_patch.core           |Subtitle formats requested: ('srt',)|
12/06/2023 12:14:11|DEBUG   |subliminal_patch.core           |Saving <EmbeddedSubtitle '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.mkv' [de:forced:utf-8]> to '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt'|
12/06/2023 12:14:11|INFO    |subliminal_patch.subtitle       |Applying mods: ['remove_HI', 'common']|
12/06/2023 12:14:11|DEBUG   |subliminal_patch.core           |Saving <OpenSubtitlesComSubtitle 'https://www.opensubtitles.com/de/subtitles/legacy/9360472' [de:forced:utf-8]> to '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt'|
12/06/2023 12:14:11|INFO    |subliminal_patch.subtitle       |Applying mods: ['remove_HI', 'common']|
12/06/2023 12:14:11|DEBUG   |subliminal_patch.core           |Saving <OpenSubtitlesComSubtitle 'https://www.opensubtitles.com/de/subtitles/legacy/9216152' [de:forced:utf-8]> to '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt'|
12/06/2023 12:14:11|INFO    |subliminal_patch.subtitle       |Applying mods: ['remove_HI', 'common']|
12/06/2023 12:14:11|DEBUG   |subliminal_patch.core           |Saving <OpenSubtitlesComSubtitle 'https://www.opensubtitles.com/de/subtitles/legacy/9216150' [de:forced:utf-8]> to '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt'|
12/06/2023 12:14:11|INFO    |subliminal_patch.subtitle       |Applying mods: ['remove_HI', 'common']|
12/06/2023 12:14:12|DEBUG   |subliminal_patch.core           |Saving <OpenSubtitlesComSubtitle 'https://www.opensubtitles.com/de/subtitles/legacy/9377025' [de:forced:utf-8]> to '/data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt'|
12/06/2023 12:14:12|INFO    |subliminal_patch.subtitle       |Applying mods: ['remove_HI', 'common']|
12/06/2023 12:14:12|DEBUG   |root                            |BAZARR Subtitles file saved to disk: /data/media/TVShows/House of the Dragon/Season 01/House of the Dragon (2022) - S01E01 - The Heirs of the Dragon [Bluray-1080p][AC3 5.1][x264]-TSCC.de.forced.srt|
```

**Comment by vitiko98** [comment]
Both providers are sending forced subtitles (bazarr is registering it in its database), so it looks like the problem is somewhere in `bazarr/subtitles`. Probably some race condition?

**Comment by morpheus65535** [comment]
@vitiko98 but I don't get why it download 5 subtitles when we ask for the best subtitles for a single language...

**Comment by vitiko98** [comment]
Probably related to c11af96950f134aca6b2331b03a206960bd883ba

**Comment by morpheus65535** [comment]
@PCJones can you confirm if your issue is fixed by v1.2.5-beta?

**Comment by morpheus65535** [comment]
No news, good news!
